src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -316,8 +327,6 @@ private async Task LoadKeyValuesRegisteredForRefresh(IDictionary<string, Configu
                     watchedKv = null;
                 }
 
-                changeWatcher.CacheExpires = DateTimeOffset.UtcNow.Add(changeWatcher.CacheExpirationInterval);

```

Comment:
>changeWatcher.CacheExpires = DateTimeOffset.UtcNow.Add(changeWatcher.CacheExpirationInterval); [](start = 16, length = 94)

I though a bit more about this line and it seems this would cause issues in the scenario when there is a change in the sentinel key with `refreshAll` set. Removing this line would cause the next refresh within 30 seconds to ignore the cached values that were just retrieved from the server.